torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <c10/core/thread_pool.h>
+#include <c10d/Store.hpp>
+#include <tensorpipe/core/context.h>
+#include <tensorpipe/core/listener.h>
+#include <tensorpipe/core/pipe.h>
+#include <torch/csrc/distributed/rpc/python_rpc_handler.h>
```

Comment:
Doesn't seem we need this header. This problem exists in `process_group_agent.h` as well. Create #35767 to remove that.